Add constants (#41)

MobiliteDev opened the pull request:
I've added all the constants, renamed the class name with "Sql" prefix.
Added to the sqlite3.dart for public visibility

MobiliteDev commented:
@simolus3 these CI errors are on your side I think.

![image](https://user-images.githubusercontent.com/54275508/121164508-eb43e900-c84f-11eb-83ab-0784289c164a.png)

simolus3 commented on sqlite3/lib/src/ffi/constants.dart:
```diff
@@ -1,44 +1,247 @@
+/// These bit values are intended for use in the
+/// 3rd parameter to the [sqlite3_open_v2()] interface and
+/// in the 4th parameter to the [sqlite3_vfs.xOpen] method.
+class SqlFlags {
```
Do we need to add those flags to the public api? I think we don't provide an api where users could pass one of them, so I think we shouldn't make them public.

MobiliteDev replied:
You are right, I've changed the visibility in sqlite3.dart on the last commit

simolus3 commented on sqlite3/lib/src/ffi/constants.dart:
```diff
@@ -1,44 +1,247 @@
 // Prepare flags, https://www.sqlite.org/c3ref/c_prepare_normalize.html
-const SQLITE_PREPARE_PERSISTENT = 0x01;
-const SQLITE_PREPARE_NO_VTAB = 0x04;
+class SqlPrepareFlags {
```
Same here I think?

MobiliteDev replied:
You are right, I've changed the visibility in sqlite3.dart on the last commit

simolus3 commented on sqlite3/lib/src/ffi/constants.dart:
```diff
@@ -1,44 +1,247 @@
+}
+
+/// Datatypes, https://sqlite.org/c3ref/c_blob.html
+class SqlTypes {
```
This one is exposed through Dart types, users shouldn't see those.

MobiliteDev replied:
You are right, I've changed the visibility in sqlite3.dart on the last commit

simolus3 commented on sqlite3/lib/src/ffi/constants.dart:
```diff
@@ -1,44 +1,247 @@
-const SQLITE_BLOB = 4;
-const SQLITE_NULL = 5;
+/// Special destructors, https://www.sqlite.org/c3ref/c_static.html
+class SqlSpecialDestructor {
```
Nice call to introduce fields for these values! Let's just not expose them to users.

MobiliteDev replied:
You are right, I've changed the visibility in sqlite3.dart on the last commit

simolus3 commented on sqlite3/lib/src/ffi/constants.dart:
```diff
@@ -1,44 +1,247 @@
+///
+/// See also: SQLITE_IOERR_READ | extended result codes,
+/// sqlite3_vtab_on_conflict() SQLITE_ROLLBACK | result codes.
+class SqlErrors {
```
Let's call this class `SqlError` which makes it easier to read a specific constant in code.

MobiliteDev replied:
You are right, I've renamed to SqlError

simolus3 reviewed (commented):
My server should be fixed now, sorry for that.

I think we should not expose all of these constants since most of them aren't actually useful for users. You can add a `show SqlError` to only expose those.

MobiliteDev commented:
I've removed the 's' from all my classes.

and I've added SqlExtError in public visibility, you get it with `sqlite3_extended_result_codes` function in `database.dart`

simolus3 commented:
I think the CI failures are due to the Flutter dev SDK not having the necessary Dart version yet. Thanks for your contribution, this looks good to me :+1:
